File: cs/caesarcipher.cs
using System;
using System.Text;

static string CaesarCipher(string text, int key)    
{   
    Console.WriteLine(key);
    Console.WriteLine(text);

    key = TrueMod(key,26);//    Console.WriteLine("k % 26 = "+ k);

    StringBuilder result = new StringBuilder();

    foreach (char chor in text)
    {
        int code = (int)chor;

        if(code >= 65 && code <= 90)
        {
            int shiftedCode = ((code - 65 + key) % 26) + 65;
            result.Append((char)shiftedCode);
        }
        else if(code >= 97 && code <= 122)
        {
            int shiftedCode = ((code - 97 + key) % 26) + 97;
            result.Append((char)shiftedCode);
        }
        else
            result.Append(chor);
    }
    return result.ToString();
}

static int TrueMod(int dividend, int divisor) => ((dividend % divisor) + divisor) % divisor;

Console.WriteLine(CaesarCipher("middle-Outza", 2));
Console.WriteLine(CaesarCipher("okffng-Qwvbc", -2));

File: py/caesar cipher.py


def caesar_cipher(s,k):

    print(k)
    print(s)
    
    k = k % 26
    print('k % 26 = ',k)
    
    result = []

    for char in s:
        code = ord(char)

        if code >= 65 and code <= 90:
            shiftedcode = ((code - 65 + k) % 26) + 65
            result.append(chr(shiftedcode))
        elif code >= 97 and code <= 122:
            shiftedcode = ((code - 97 + k) % 26) + 97
            result.append(chr(shiftedcode))
        else:
            result.append(char)

    return "".join(result)

print(caesar_cipher('middle-Outza', -2))